src/vstest.console/CommandLine/Executor.cs

```diff
@@ -114,6 +115,24 @@ internal int Execute(params string[] args)
                 exitCode = 1;
             }
 
+            if (!isDiag)
+            {
+                var diag = Environment.GetEnvironmentVariable("VSTEST_DIAG");
+                var diagVerbosity = Environment.GetEnvironmentVariable("VSTEST_DIAG_VERBOSITY");
+                if (!string.IsNullOrWhiteSpace(diag))
+                {
+                    var verbosity = TraceLevel.Verbose;
+                    if (diagVerbosity != null)
+                    {
+                        if (Enum.TryParse<TraceLevel>(diagVerbosity, ignoreCase: true, out var parsedVerbosity))
+                        {
+                            verbosity = parsedVerbosity;
+                        }
+                    }
+                    args = args.Concat(new[] { $"--diag:{diag};TraceLevel={verbosity}" }).ToArray();
```

Comment:
The parameter is actually called TraceLevel, which was a bit surprising to me. Because all the other places I can think about use Verbosity, and imho we should standardize on that? So the env variable name now differs, because I won't be able to remember it is called trace level when I want to use it (and I don't have a place to log it into because logs are not  setup at that time.).

Reply:
Agree, I was also convinced about verbosity